conanfile.py

```diff
@@ -19,29 +19,32 @@ class Recipe(ConanFile):
 
     # Binary configuration
     settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
+    options = {"shared": [True, False], "fPIC": [True, False], "with_test_deps": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "with_test_deps": False}
     exports = "LICENSE",
     exports_sources = "src/*", "private/*", "CMakeLists.txt", "cmake/*"
-    requires = (("fmt/9.0.0", "private"),  # format must only be used within cpp files
-                "expected-lite/0.6.2",
-                "boost/1.81.0",
-                "re2/20221201",
-                "openssl/3.0.8",
-                "zlib/1.2.12",  # force override version from openssl and boost
-                "uni-algo/0.7.1")
 
     generators = ("CMakeDeps", "CMakeToolchain")
 
+    def requirements(self):
+        self.requires("boost/1.81.0")
+        self.options["boost"].header_only = True
+        self.requires("expected-lite/0.6.2")
+        self.requires("re2/20221201")
+        self.requires("openssl/3.0.8")
+        self.requires("uni-algo/0.7.1")
+
+        if self.options.with_test_deps:
+            self.requires("doctest/2.4.11")
+
+
     def set_version(self):
         if not hasattr(self, 'version') or self.version is None:
             cmake_file = load(self, os.path.join(self.recipe_folder, "CMakeLists.txt"))
-            self.version = re.search(r"project\([^)]*VERSION\s+(\d+\.\d+.\d+)[^)]*\)", cmake_file).group(1)
```

Comment:
1. why did you remove this line? 
2. As you removed this line, why did you not remove set_version entirely?

Reply:
That must have been an accident, at least I don't remember doing that intentionally.